test/suite/db_clean.go

```diff
@@ -62,21 +65,22 @@ func DeleteCreatedEntities(db *gorm.DB) func() {
 		if !inTransaction {
 			tx = db.Begin()
 		}
-		for i := len(entires) - 1; i >= 0; i-- {
-			entry := entires[i]
+		for i := len(entities) - 1; i >= 0; i-- {
+			entity := entities[i]
 			log.Debug(nil, map[string]interface{}{
-				"table":     entry.table,
-				"key":       entry.key,
+				"table":     entity.table,
+				"keys":      entity.keys,
 				"hook_name": hookName,
-			}, "Deleting entities from '%s' table with key %v", entry.table, entry.key)
-			tx.Table(entry.table).Where(entry.keyname+" = ?", entry.key).Delete("")
+			}, "Deleting entities from '%s' table with keys %v", entity.table, entity.keys)
+			if len(entity.keys) == 0 {
+				log.Panic(nil, map[string]interface{}{
+					"table":     entity.table,
+					"hook_name": hookName,
+				}, "no primary keys found!!!")
+			}
+			tx.Table(entity.table).Where(entity.keys).Delete("")
```

Comment:
we should check the error to make sure that the deletion failed. For now, the errors are silently ignored and this explains why we have left-over records in the database without even realizing it. We should definitely make sure that all records are correctly removed.

Reply:
Yeah.. good point. But do you think we should fail here? If we just log it then it won't be visible (especially if tests pass) and it's good as ignoring these errors.

Reply:
I think we should fail, in order to make sure that the DB is cleaned afterwards. But I also understand that this may cause unexpected failures on some existing tests. So maybe we could log the error but only fail if a specific/dedicated env variable is set (or not fail if a specific env var is set?)

Reply:
because frankly, left-over records from previous tests are one of the main reasons that a test suddenly fails, and it's annoying

Reply:
Done in https://github.com/fabric8-services/fabric8-common/pull/65/commits/c2ec459ebae3540425cd5fb99034e85875c6f00b